File: src/TwelveWeekYearWebSpa/src/app/goals/models/goal.ts
export enum GoalTypes {
  None = 0,
  ThreeToFiveYear = 1,
  ThisYear = 2,
  TwelveWeekYear = 3
}
export interface SubTask {
  id: string;
  description: string;
}

export interface Task {
  id: string;
  description: string;
  subTasks: SubTask[];
}

export interface Goal {
  id: string;
  description: string;
  type: GoalTypes;
  tasks: Task[];
}

File: src/TwelveWeekYear.Domain/Models/Goal.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace TwelveWeekYear.Domain.Models
{
	public class Goal
	{
		[Key]
		public int Id { get; set; }
		[Required]
		public string Name { get; set; }
		[Required]
		public string Description { get; set; }

		[Required]
		public int GoalTypeId { get; set; }
		public GoalType GoalType { get; set; }

		public ICollection<Task> Tasks { get; set; } = new List<Task>();

		//[Required]
		public int TweleveWeekYearId { get; set; }

		public TweleveWeekYear TweleveWeekYear { get; set; }
	}
}
